Sources/TensorFlow/Operators/NN.swift

```diff
@@ -821,6 +821,86 @@ func _vjpAvgPool3D<Scalar: TensorFlowFloatingPoint>(
   )
 }
 
+/// Returns a 2-D fractional max pooling, with the specified pooling ratios.
+///
+/// Note: `fractionalMaxPool` does not have an XLA implementation, and thus may have performance implications.
+///
+/// - Parameters:
+///   - input: A Tensor. 4-D with shape `[batch, height, width, channels]`.
+///   - poolingRatio: A list of `Doubles`. Pooling ratio for each dimension of `input`, currently only
+///     supports row and col dimension and should be >= 1.0.
+///   - pseudoRandom: An optional `Bool`. Defaults to `false`. When set to `true`,
+///     generates the pooling sequence in a pseudorandom fashion, otherwise, in a random fashion.
+///   - overlapping: An optional `Bool`. Defaults to `false`. When set to `true`, it means
+///     when pooling, the values at the boundary of adjacent pooling cells are used by both cells.
+///   - deterministic: An Optional `Bool`. When set to `true`, a fixed pooling region will be
+///     used when iterating over a fractionalMaxPool node in the computation graph.
+///   - seed: An optional `Int64`. Defaults to `0`. If set to be non-zero, the random number
+///     generator is seeded by the given seed.
+///   - seed2: An optional `Int64`. Defaults to `0`. A second seed to avoid seed collision.
+@differentiable(wrt: input)
+public func fractionalMaxPool<Scalar: TensorFlowFloatingPoint>(
```

Comment:
`fractionalMaxPool` supports only 2D input.
It should be `fractionalMaxPool2D` like `maxPool2D`.

Reply:
Done